2022/Day7/RFC.md

```diff
@@ -0,0 +1,87 @@
+# AoC 2022 Day 6
+## Overview:
+You can hear birds chirping and raindrops hitting leaves as the expedition proceeds. Occasionally, you can even hear 
+much louder sounds in the distance; how big do the animals get out here, anyway?
+
+The device the Elves gave you has problems with more than just its communication system. You try to run a system update:
+
+```
+$ system-update --please --pretty-please-with-sugar-on-top
+Error: No space left on device
+```
+
+Perhaps you can delete some files to make space for the update?
+
+The filesystem consists of a tree of files (plain data) and directories (which can contain other directories or files). 
+The outermost directory is called /. You can navigate around the filesystem, moving into or out of directories and 
+listing the contents of the directory you're currently in.
+
+Since the disk is full, your first step should probably be to find directories that are good candidates for deletion. 
+To do this, you need to determine the total size of each directory. The total size of a directory is the sum of the 
+sizes of the files it contains, directly or indirectly. (Directories themselves do not count as having any intrinsic size.)
+
+## Glossary:
+
+## Part 2:
+### Goal:
+The total disk space available to the filesystem is 70000000. To run the update, you need unused space of at least 30000000. 
+You need to find a directory you can delete that will free up enough space to run the update.
+
+Find the smallest directory that, if deleted, would free up enough space on the filesystem to run the update. 
+What is the total size of that directory?
+
+### Design:
+#### Parsing:
+No additional parsing needed.
+
+#### Solution:
+The first thing to do is to find how much space we need to free up to get the requirement. We can do that by summing all
+the directories and subtracting that from the total disk space available.
+
+Now that we have our limit, we can simply add a new method to the `Directory` class, that finds the smallest directory to
+be deleted that would free up enough space, by finding every directory's size that is larger or equal to the limit, and 
```

Comment:
Finding how?  A recursive search that yields the appropriate directories?  A recursive search that somehow only returns the smallest size larger than x?

Reply:
This would be similar to the `sum_all_subdirectories_under_limit` method except instead of under its over the limit. This would be a recursive search that would create a list of all directories over the limit. It would then take the min of that list.

Reply:
But... sum_all_subdirectories_under_limit _doesn't_ create a list of all directories under the limit?

Reply:
It doesn't. It creates a sum. I'd likely call the other one a better name so there was a difference, but the high level logic would be the same.

Reply:
I... don't think it would be.  There are some pretty significant differences to accumulation vs alternation.  You might want to be more clear about it.  Importantly, the words you're saying here don't reflect your implementation at all.

Reply:
That's fair - and I didn't rewrite this yet because I wanted to talk it out in the comments, before making the change.